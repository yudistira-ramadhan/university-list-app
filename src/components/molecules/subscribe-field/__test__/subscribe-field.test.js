import React from 'react'
import ReactDOM from 'react-dom'
import SubscribeField from '../subscribe-field'
import { render, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')

  ReactDOM.render(<SubscribeField />, div)
})

it('renders subscribe field correctly', () => {
  const { getByTestId, getByPlaceholderText } = render(<SubscribeField />)

  expect(getByTestId('subscribe-button')).toBeTruthy()

  expect(getByPlaceholderText('Enter your email...')).toBeTruthy()
})

describe('input value', ()=> {
  it('updates on change', ()=> {

    const { getByPlaceholderText } = render(<SubscribeField/>)

    const emailInput = getByPlaceholderText('Enter your email...')

    fireEvent.change(emailInput, { target: { value: 'test@testmail.com' } })

    expect(emailInput.value).toBe('test@testmail.com')
  })
})

describe('search button', ()=>{
  describe('with empty value', ()=> {
    it('does not trigger request search function', ()=> {
      const onHandleClick = jest.fn()

      const { getByTestId } = render(<SubscribeField onHandleClick={ onHandleClick } />)

      fireEvent.click(getByTestId('subscribe-button'))

      expect(onHandleClick).not.toHaveBeenCalled()
    })
  })

  describe('with value inside field', ()=>{
    describe('with non email value', () => {
      it('does not trigger onHandleClick function', ()=>{
        const onHandleClick = jest.fn()
        global.URL.createObjectURL = jest.fn()
  
        const { getByTestId, getByPlaceholderText } = render(<SubscribeField onHandleClick={ onHandleClick } />)
  
        const emailInput = getByPlaceholderText('Enter your email...')
  
        fireEvent.change(emailInput, { target: { value: 'test' } })
  
        fireEvent.click(getByTestId('subscribe-button'))
  
        expect(onHandleClick).not.toHaveBeenCalled()
      })
    })
  })
})

it('matches snapshot', () => {
  const tree = renderer.create(<SubscribeField />).toJSON()
  expect(tree).toMatchSnapshot()
})