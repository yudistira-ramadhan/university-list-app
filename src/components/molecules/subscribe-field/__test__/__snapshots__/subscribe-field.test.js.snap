// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className=""
>
  <div
    className="form-group"
  >
    <div
      className="row"
    >
      <div
        className="pr-0 col-7 col-sm-9 col-md-9"
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Enter your email..."
          type="email"
        />
      </div>
      <div
        className="col-5 col-sm-3 col-md-3"
      >
        <button
          aria-label={null}
          className="py-1 w-100 btn btn-outline-primary"
          data-testid="subscribe-button"
          onClick={[Function]}
          type="button"
        >
          SUBSCRIBE
        </button>
      </div>
    </div>
  </div>
</form>
`;
